packet.go

```diff
@@ -314,5 +396,107 @@ func (p *Packet) decodeIp6(recur int) {
 		p.decodeIcmp()
 	case IP_INIP:
 		p.decodeIp6(recur + 1)
+	case IP_GRE:
+		p.decodeGre()
+	}
+}
+func (p *Packet) decodeVxLAN() bool {
+	// VxLAN decode is a heuristic and not an exact algorithm.
+
+	// VXLAN Header is 8 bytes:
+	//  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	// |R|R|R|R|I|R|R|R|            Reserved                           |
+	// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	// |                VXLAN Network Identifier (VNI) |   Reserved    |
+	// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	if len(p.Payload) < 8 {
+		return false
+	}
+	pkt := p.Payload
+
+	// if 'I' bit is not set, there is no valid VNI.
+	// Which means it is very less likely to be a vxlan
+	// packet. We can safely return.
+	if uint8(pkt[0])&0x08 == 0 {
```

Comment:
If other bits are set too we can ignore this from rfc so may be == 0x80

VXLAN Header:  This is an 8-byte field that has:

      - Flags (8 bits): where the I flag MUST be set to 1 for a valid
        VXLAN Network ID (VNI).  The other 7 bits (designated "R") are
        reserved fields and MUST be set to zero on transmission and
        ignored on receipt.

Reply:
For the reception case the other flags might be valid, so I think the last comment an be ignored.